sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/LogExporterConfigurationTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.collect.ImmutableMap;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.exporter.otlp.logs.OtlpGrpcLogExporter;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
+import io.opentelemetry.sdk.logs.export.LogExporter;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+
+public class LogExporterConfigurationTest {
+
+  @Test
+  void configureOtlpLogsUnsupportedProtocol() {
```

Comment:
I don't think this is the right place to test this. We're mixing concerns here of validating that system properties can translated to the correct `setCompression` calls, that `setCompression` results in the correct behavior on the resulting exporter, and that both the global and signal specific version of the system properties behave as intended. 

I think we need to focus specifically on ensuring that calls to `setCompression` result in the correct behavior on the resulting exporter, since the other two concerns already have test coverage elsewhere. Can we extend [AbstractGrpcTelemetryExporterTest](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/AbstractGrpcTelemetryExporterTest.java) instead? If so, we'll have a small amount of test code that covers all versions of the OTLP exporters.

Reply:
Nice, that looks good.

> Should I remove the new test methods from MetricExporterConfigurationTest and SpanExporterConfigurationTest as well? It feels wrong to not test the new calls to setCompression() in SpanExporterConfiguration at all.

The autoconfigure test situation has gotten a bit unwieldy as there are overlapping tests and its become unclear where to tests particular elements. Frankly, I think the whole lot of tests needs a refactor. However, for the purposes of this PR, there is already coverage of setting `gzip` compression in `OtlpHttpConfigTest` and `OtlpGrpcConfigTest`. Obviously the coverage wasn't completely sufficient or the bugs you uncovered wouldn't have existed! I'd recommend adding very focussed tests to `OtlpConfigUtilTest`: `OtlpConfigUtil` is responsible for parsing OTLP config properties and calling the appropriate setters on exporter builders. If we test that the we properly extract the compression properties, and separately test that those properties make it to the exporter builders, we should have tight coverage.

Reply:
I will move the OTLP related tests to `OtlpConfigUtilTest`, but leave the tests for Jaeger and Zipkin in `SpanExporterConfigurationTest`. But as agreed, those changes will anyway be dropped for now.